src/components/v2/TextField/index.tsx

```diff
@@ -16,18 +16,41 @@ export interface ITextFieldProps extends Omit<InputHTMLAttributes<HTMLInputEleme
 }
 
 export const TextField: React.FC<ITextFieldProps> = ({
-  css,
+  className,
   label,
   description,
   hasError = false,
   leftIconName,
   rightAdornment,
+  onChange,
+  max,
+  min,
+  type,
   ...inputProps
 }) => {
   const styles = useStyles();
 
+  const handleChange: InputHTMLAttributes<HTMLInputElement>['onChange'] = e => {
+    // Prevent value from being updated if it does not follow the rules
+    const followsMaxRule =
```

Comment:
I don't think we should put validation in the TextComponent. I think we should put it in customized components for specific validation and text types. Ideally validation wouldn't happen in the Text component at all but would happen at the form level using a library like `yup`

Reply:
So in this case it's a bit different, because we're actually blocking the user's action when these rules aren't followed.
As far as I can see, `yup` wouldn't let us do that as it would validates the input after it's been entered already.
I agree validation should be outside the `TextField` component, however I was under the impression this doesn't quite fall in that category since it is blocking a user action. I'm happy to make any modifications we see fit.
What's your opinion on that?

Reply:
ok I understand, in that case if we have start to see lots of diffenent logic and styles for number and string inputs we might want to break them out but in that context this makes sense